sshd-core-0.14.0/src/org/apache/sshd/server/sftp/SftpSubsystem.java

```diff
@@ -601,10 +601,6 @@ protected void process(Buffer buffer) throws IOException {
                     } else {
                         FileHandle fh = (FileHandle) p;
                         fh.write(data, offset);
-                        SshFile sshFile = fh.getFile();
-
-                        sshFile.setLastModified(new Date().getTime());
```

Comment:
Is it necessary to remove that?
When/where is setLastModified() invoked?

Reply:
It's not necessary strictly, but slows down the copy, it calls this after each write block. Depending on the client (block size 4k-64k), 100 or 1000 times during each a file upload. :(

When invoked? In case of WinSCP:
- If I turn off "preserve timestamp", WinSCP calls only open/write/close, and the OS sets the last modified time.
- If I enable "preserve timestamp", it send a separate SSH_FXP_SETSTAT message, and that sets the timestamp.
- Tested with SAF and POFS (plain old FS) and reading the ftpd logs.